src/interfaces.js

```diff
@@ -73,5 +73,65 @@ interface RouteConfig {
   */
   caseSensitive?: boolean;
 
+  /**
+  * Add to specify an activation strategy if it is always the same and you do not want that
+  * to be in your view-model code. Available values are 'replace' and 'invoke-lifecycle'.
+  */
+  activationStrategy?: string;
+  
   [x: string]: any;
 }
+
+type ShouldContinue = boolean|Promise<boolean>|PromiseLike<boolean>|NavigationCommand;
+type Potential = Promise<void>|PromiseLike<void>|IObservable;
```

Comment:
Not much for me, but I'll add it for completeness and future references. But as `Potential|any` since the 2 `if`s exhaust `Potential` and in ts 2.0, after the last `if` the type will be `never`. I'll wait for any other input and if non any, I'll squash the commits or I shouldn't?